data-azure-cosmos/src/main/java/io/micronaut/data/cosmos/operations/DefaultCosmosRepositoryOperations.java

```diff
@@ -389,4 +384,120 @@ private <E, R> SqlStoredQuery<E, R> getSqlStoredQuery(StoredQuery<E, R> storedQu
         }
         throw new IllegalStateException("Expected for prepared query to be of type: SqlStoredQuery got: " + storedQuery.getClass().getName());
     }
+
+    private void logQuery(SqlQuerySpec querySpec, Iterable<SqlParameter> params) {
+        if (QUERY_LOG.isDebugEnabled()) {
+            QUERY_LOG.debug("Executing query: {}", querySpec.getQueryText());
+            for (SqlParameter param : params) {
+                QUERY_LOG.debug("Parameter: name={}, value={}", param.getName(), param.getValue(Object.class));
+            }
+        }
+    }
+
+    // Init database and containers
+
+    private CosmosDatabase initDatabase(CosmosDatabaseConfiguration configuration) {
+        CosmosDatabaseResponse databaseResponse;
+        ThroughputProperties throughputProperties = createThroughputProperties(configuration);
+        if (throughputProperties == null) {
+            databaseResponse = cosmosClient.createDatabaseIfNotExists(configuration.getDatabaseName());
+        } else {
+            databaseResponse = cosmosClient.createDatabaseIfNotExists(configuration.getDatabaseName(), throughputProperties);
+        }
+        CosmosDatabase cosmosDatabase = cosmosClient.getDatabase(databaseResponse.getProperties().getId());
+        initContainers(cosmosDatabase);
+        return cosmosDatabase;
+    }
+
+    private ThroughputProperties createThroughputProperties(CosmosDatabaseConfiguration configuration) {
+        // Throughput properties for the database
+        if (configuration.getThroughputRequestUnits() != null) {
+            if (configuration.isThroughputAutoScale()) {
+                return ThroughputProperties.createAutoscaledThroughput(configuration.getThroughputRequestUnits());
+            } else {
+                return ThroughputProperties.createManualThroughput(configuration.getThroughputRequestUnits());
+            }
+        }
+        return null;
+    }
+
+    private void initContainers(CosmosDatabase cosmosDatabase) {
```

Comment:
Please extract this code in the future to some Cosmos Initializer. It should be disabled by default.